### ISSUE TITLE
System error: not a directory

### ISSUE THREAD

**Issue by SultanICQ** [deviantony/docker-elk#23]
Hi, I ran into a problem when I was working with your images.

After all downloads and buildings I got this error.

```bash
Creating dockerelkmaster_kibana_1
ERROR: Cannot start container 3a4721181937361993ea01bb7bb350834acaf080c3a40c1330d91cf5848234eb: [8] System error: not a directory
```

I tried several modifications to your docker-compose.yml file and the problem seems to be located with the volumes section. If I comment out these lines the process continues.

```yml
#  volumes:
#    - ./kibana/config/kibana.yml:/opt/kibana/config/kibana.yml
```

But your costumizations didn't get applied.

I'm using docker-compose version: 1.5.1 and Docker version 1.9.0, build 76d6bc9

Is this normal?

**Comment by deviantony** [comment]
I've just upgraded docker to 1.9.1 and docker-compose to 1.5.1 and I started the stack without any issues.

```bash
$ docker info                                                                                                                                                                                                                        
Server Version: 1.9.1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 411
 Dirperm1 Supported: false
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.5.0-61-generic
Operating System: Ubuntu precise (12.04.5 LTS)
```

Seems to be related to these issues: https://github.com/docker/compose/issues/424 and https://github.com/docker/compose/issues/2268

**Comment by SultanICQ** [comment]
Hi!

I found the problem. I installed docker-osx-dev to sync host and docker folders and it wants to remove the /Users shared folder inside the docker machine.

DockerMachine can't access the folder without that shared folder.

Once reshared it worked again.

Thankyou!

**Comment by deviantony** [comment]
So it's related to docker-machine? Could you describe your solution a quickly in this issue? I'll let it open so other people will be able to check the workaround.

**Comment by jlgoolsbee** [comment]
I'm also running into this issue "out of the box" on Mac OS 10.11 using docker-machine:

```
$ docker-machine --version
docker-machine version 0.5.1 (HEAD)
```
_Admittedly, docker-machine 0.5.2 is out but I manage my docker environment with Homebrew, and the 0.5.2 update isn't quite ready yet (https://github.com/Homebrew/homebrew/pull/46598); I will happily re-test when it is._

```
$ docker -v
Docker version 1.9.1, build a34a1d5
```

```
$ docker info
Containers: 4
Images: 77
Server Version: 1.9.1
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 85
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 4.1.13-boot2docker
Operating System: Boot2Docker 1.9.1 (TCL 6.4.1); master : cef800b - Fri Nov 20 19:33:59 UTC 2015
CPUs: 1
Total Memory: 996.2 MiB
Name: default
Debug mode (server): true
 File Descriptors: 13
 Goroutines: 20
 System Time: 2015-12-07T21:46:39.098642519Z
 EventsListeners: 0
 Init SHA1: 
 Init Path: /usr/local/bin/docker
 Docker Root Dir: /mnt/sda1/var/lib/docker
Registry: https://index.docker.io/v1/
Labels:
 provider=virtualbox
```

```
$ docker-compose --version
docker-compose version 1.5.2, build unknown
```

```
$ docker-compose up
Starting dockerelk_elasticsearch_1
Starting dockerelk_logstash_1
Creating dockerelk_kibana_1
ERROR: Cannot start container 44153d4ee3edfe26eea92580a63545385468ff178124662c68afce986b43cb51: [8] System error: not a directory
```

It would seem that https://github.com/deviantony/docker-elk/issues/22 and this issue are related, and if so, I guess neither are actually an issue with this project – I guess you might just need to update the documentation if/when there's an actual resolution – but consider this a fairly verbose "+1".

**Comment by deviantony** [comment]
@jlgoolsbee Did you try the solution proposed in #22 ?

**Comment by lucj** [comment]
Hi, I still do not manage to run docker-elk on a docker host created with Machine. I have the same 'not a directory' error:   

     ERROR: Cannot start container xxxxxxxx: [8] System error: not a directory

Default path of the host

    $ docker-machine ssh elk1 pwd
    /home/docker

I have copied over the docker-elk directory to the host

    $ docker-machine ssh elk1 ls -lrt
    total 20
    -rw-r--r--    1 docker   staff           69 Dec 16 12:50 log.log
    drwxr-xr-x    5 docker   staff          160 Dec 16 13:31 docker-elk
    drwxr-xr-x    3 docker   staff           60 Dec 16 13:35 logstash
    drwxr-xr-x    3 docker   staff          100 Dec 16 13:35 kibana
    drwxr-xr-x    3 docker   staff           60 Dec 16 13:35 elasticsearch
    -rw-r--r--    1 docker   staff          501 Dec 16 13:35 docker-compose.yml
    -rw-r--r--    1 docker   staff         6314 Dec 16 13:35 README.md
    -rw-r--r--    1 docker   staff         1082 Dec 16 13:35 LICENSE

Any idea what I'm missing ?

**Comment by lucj** [comment]
Ok, typo on my side.. I did not change the path of the docker-compose file correctly.
After making sure the volumes uses /home/docker/docker-elk, it's working fine.
Thanks a lot for this repo.

**Comment by jamesabbottsmith** [comment]
I've hit a similar issue. I'm not an expert on Docker (yet!), but I'm wondering about the wisdom in using volumes for passing configuration files in the first place (which is the source of this issue). 

Shouldn't the configuration be added (ADD / COPY) in the Dockerfile so that the images contain their configuration? If I have to separately copy the configuration files to the host, then I've lost the configuration management inherent with Docker.

**Comment by deviantony** [comment]
The philosophy around this project was to use the official ELK images without bloating them too much.

That's why I've preferred mount the configuration files instead of creating a new Dockerfile for each component to COPY the configuration. The other good point of this practice is that you can update the configuration file and restart the container to see your changes applied, without having to rebuild the image.

"If I have to separately copy the configuration files to the host, then I've lost the configuration management inherent with Docker."

Again, I thought that the aim of this project was not to ship new ELK images (with their configuration), but re-use the official ones.

**Comment by lucj** [comment]
+1 for the use of official ELK images

On Tue, Dec 22, 2015 at 8:33 AM, Anthony Lapenna <notifications@github.com>
wrote:

> The philosophy around this project was to use the official ELK images
> without bloating them too much.
>
> That's why I've preferred mount the configuration files instead of
> creating a new Dockerfile for each component to COPY the configuration. The
> other good point of this practice is that you can update the configuration
> file and restart the container to see your changes applied, without having
> to rebuild the image.
>
> "If I have to separately copy the configuration files to the host, then
> I've lost the configuration management inherent with Docker."
>
> Again, I thought that the aim of this project was not to ship new ELK
> images (with their configuration), but re-use the official ones.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/deviantony/docker-elk/issues/23#issuecomment-166541977>
> .
>



-- 
Luc Juggery - https://about.me/lucjuggery

**Comment by jamesabbottsmith** [comment]
Thanks for your response @deviantony. I do appreciate the fact that this repo isn't bloated. For my purposes of opted to modify it slightly so that I do rebuild kibana + logstash so I can COPY configuration in. It does require me to `docker-compose build` before `docker-compose up`, but it saves me copying the configuration files to the remote docker host as in #22

**Comment by geniousphp** [comment]
Any update guys? Will Docker-compose allow mounting individual files in the future?

**Comment by deviantony** [comment]
@geniousphp It's already working with docker-compose, see https://github.com/docker/compose/issues/424

The problem here seems to be related to docker-machine.

**Comment by geniousphp** [comment]
Thanks @deviantony. There's still a little issue with docker-compose. I'll post it in docker/compose#424

**Comment by funkytaco** [comment]
I'm getting `ERROR: Cannot start container 8eecd7...: [8] System error: not a directory` on Docker Swarm.

>That's why I've preferred mounting the configuration files rather than creating a new Dockerfile for each component and COPY the configuration. 

Well, I'm going to use this in a swarm, not on a docker-machine, so I may just fork the project.

**Comment by lucj** [comment]
I've started a little fork with Anthony's shield-support, maybe that could help: https://github.com/lucj/docker-elk/tree/shield-support

**Comment by deviantony** [comment]
I've juste merged #42 which should solve this problem.
